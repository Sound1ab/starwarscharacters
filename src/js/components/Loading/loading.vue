<template>
	<transition name="fade-up" mode="out-in">
		<div class="loading" v-if="loading">
			<img class="loading__c3po" :src="require('@/assets/c3po.png')" />
		</div>
	</transition>
</template>

<script>
	import {mapState} from 'vuex';
	export default {
		name: 'loading',
		computed: {
			...mapState({
				loading: state => state.ui.loading
			})
		}
	};
</script>

<style lang="scss" type="text/scss">
	.loading {
		position: absolute;
		top: 0;
		left: 0;
		width: 100%;
		height: 100%;
		display: flex;
		justify-content: center;
		align-items: center;
		&__c3po{
			width: 250px;
			height: 250px;
		}
	}
</style>
