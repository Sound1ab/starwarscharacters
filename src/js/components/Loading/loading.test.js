/* eslint-disable */
import {shallow, createLocalVue} from '@vue/test-utils';
import {createRenderer} from 'vue-server-renderer';
import Vuex from 'vuex';
import Loading from './loading';

const localVue = createLocalVue();
localVue.use(Vuex);

let mountComponent = (store) => shallow(Loading, {
	localVue,
	store
});

describe('Loading', () => {
	let wrapper;
	let store;
	beforeEach(() => {
		store = new Vuex.Store({
			state: {
				ui: {
					loading: true
				}
			}
		});
		wrapper = mountComponent(store);
		jest.clearAllMocks();
	});
	afterEach(() => {
		wrapper.destroy();
	});
	describe('htmlStructure', () => {
		it('html snapshot matches', () => {
			expect(wrapper.html()).toMatchSnapshot();
		});
	});
	describe('component', () => {
		it('vm - is instantiated', () => {
			expect(wrapper.isVueInstance()).toBe(true);
		});
	});
	describe('interface', () => {
		it('should show loading icon when loading state is true', () => {
			expect(wrapper.is('div')).toBe(true);
			wrapper.vm.$store.state.ui.loading = false;
			expect(wrapper.isEmpty()).toBe(true);
		});
	});
});
