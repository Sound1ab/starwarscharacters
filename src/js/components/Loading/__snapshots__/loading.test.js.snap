// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading htmlStructure html snapshot matches 1`] = `
<div class="loading" name="fade-up" mode="out-in">
    <img src="[object Object]" class="loading__c3po">
</div>
`;
